internal/authservice/service.go

```diff
@@ -397,6 +398,31 @@ func (s *Service) samlAcs(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// if the saml flow was created as part of the admin test mode, then
+	// redirect to the admin ui
+	if createSAMLLoginRes.SAMLFlowTestModeIDP != "" {
+		redirectURL, err := url.Parse(dataRes.EnvironmentAdminTestModeURL)
+		if err != nil {
+			panic(err)
+		}
+
+		attributes, err := json.Marshal(validateRes.SubjectAttributes)
+		if err != nil {
+			panic(err)
+		}
+
+		redirectQuery := url.Values{}
+		redirectQuery.Set("idp", createSAMLLoginRes.SAMLFlowTestModeIDP)
+		redirectQuery.Set("saml_connection_id", samlConnID)
+		redirectQuery.Set("email", email)
+		redirectQuery.Set("attributes", string(attributes))
+		redirectURL.RawQuery = redirectQuery.Encode()
+		redirect := redirectURL.String()
+
+		http.Redirect(w, r, redirect, http.StatusSeeOther)
+		return
+	}
+
```

Comment:
This is the substantive place where we change the default behavior in auth.ssoready.com.